paddle/fluid/operators/conv_op.cc

```diff
@@ -290,20 +288,11 @@ void ConvOpGrad::InferShape(framework::InferShapeContext* ctx) const {
 
 framework::OpKernelType ConvOpGrad::GetExpectedKernelType(
     const framework::ExecutionContext& ctx) const {
-  bool use_cudnn = ctx.Attr<bool>("use_cudnn");
-  use_cudnn &= platform::is_gpu_place(ctx.GetPlace());
-#ifdef PADDLE_WITH_CUDA
-  if (platform::is_gpu_place(ctx.GetPlace())) {
-    auto& dev_ctx = ctx.template device_context<platform::CUDADeviceContext>();
-    use_cudnn &= dev_ctx.cudnn_handle() != nullptr;
-  }
-#endif
-
-  framework::LibraryType library_;
-  if (use_cudnn) {
+  framework::LibraryType library_{framework::LibraryType::kPlain};
+  if (CanCUDNNBeUsed(ctx)) {
     library_ = framework::LibraryType::kCUDNN;
-  } else {
-    library_ = framework::LibraryType::kPlain;
+  } else if (CanMKLDNNBeUsed(ctx)) {
+    library_ = framework::LibraryType::kMKLDNN;
   }
 
   std::string data_format = ctx.Attr<std::string>("data_format");
```

Comment:
Here we only make MKLDNN library enabled.

As synced with @pzelazko-intel, we would enable MKLDNN layout next time.
Then we would considerate transform function as well.

Reply:
Could you add "TODO" in codes for reminding? Including:
- enable MKLDNN layout
- enable groups
- something more.

Besides, could you not cover the previous commit next time, since we could not find the difference after your updating?

Reply:
OK, I'm going to add TODOs.
I've covered previous commits, because I wanted commit history to be clear.
Would I have opportunity to squash commits when merging?

Reply:
Yes, we can squash commits when merging your PR.